test/gitlab_merge_request_test.go

```diff
@@ -42,25 +45,48 @@ func TestGitlabMergeRequest(t *testing.T) {
 	assert.NilError(t, err)
 
 	targetRefName := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pac-e2e-test")
-	title := "TestMergeRequest - " + targetRefName
-	err = tgitlab.PushFilesToRef(glprovider.Client, title,
-		projectinfo.DefaultBranch,
-		targetRefName,
-		opts.ProjectID,
-		entries, ".tekton/subdir/pr.yaml")
+
+	gitCloneURL, err := scm.MakeGitCloneURL(projectinfo.WebURL, opts.UserName, opts.Password)
 	assert.NilError(t, err)
+	scmOpts := &scm.Opts{
+		GitURL:        gitCloneURL,
+		Log:           runcnx.Clients.Log,
+		WebURL:        projectinfo.WebURL,
+		TargetRefName: targetRefName,
+		BaseRefName:   projectinfo.DefaultBranch,
+	}
+	scm.PushFilesToRefGit(t, scmOpts, entries)
+
 	runcnx.Clients.Log.Infof("Branch %s has been created and pushed with files", targetRefName)
-	mrID, err := tgitlab.CreateMR(glprovider.Client, opts.ProjectID, targetRefName, projectinfo.DefaultBranch, title)
+	mrTitle := "TestMergeRequest - " + targetRefName
+	mrID, err := tgitlab.CreateMR(glprovider.Client, opts.ProjectID, targetRefName, projectinfo.DefaultBranch, mrTitle)
 	assert.NilError(t, err)
 	runcnx.Clients.Log.Infof("MergeRequest %s/-/merge_requests/%d has been created", projectinfo.WebURL, mrID)
 	defer tgitlab.TearDown(ctx, t, runcnx, glprovider, mrID, targetRefName, targetNS, opts.ProjectID)
-	wait.Succeeded(ctx, t, runcnx, opts, "Merge Request", targetNS, 1, "", title)
+
+	// updating labels to test if we skip them, this used to create multiple PRs
+	_, _, err = glprovider.Client.MergeRequests.UpdateMergeRequest(opts.ProjectID, mrID, &clientGitlab.UpdateMergeRequestOptions{
+		Labels: &clientGitlab.Labels{"hello-label"},
+	})
+	assert.NilError(t, err)
+
+	// Send another Push to make an update and make sure we react to it
+	// this used to be not working
+	entries, err = payload.GetEntries(map[string]string{
+		"hello-world.yaml": "testdata/pipelinerun.yaml",
+	}, targetNS, projectinfo.DefaultBranch,
+		options.PullRequestEvent, map[string]string{})
+	assert.NilError(t, err)
+	scmOpts.BaseRefName = targetRefName
+	scm.PushFilesToRefGit(t, scmOpts, entries)
+	wait.Succeeded(ctx, t, runcnx, opts, "Merge Request", targetNS, 4, "", mrTitle)
```

Comment:
This won't work 

because https://github.com/chmouel/pipelines-as-code/blob/13a970a2078f4ec8fc896b44efb1c3114d649e39/test/pkg/wait/check.go#L35 the for loop executes only when `numberofprmatch` is `1` :grin:

Reply:
what do you mean it won't work?

Reply:
ah yeah that for loop has never worked, you are right!

Reply:
I fixed this in another commit on this PR